src/tests/EpicsConnectionStatusTests.cpp

```diff
@@ -0,0 +1,166 @@
+using nlohmann::json;
+
+static std::unique_ptr<std::int8_t[]>
+GenerateFlatbufferData(size_t &DataSize, const uint64_t Timestamp,
+                       const EventType Status, const std::string &SourceName) {
+  flatbuffers::FlatBufferBuilder builder;
+
+  auto Source = builder.CreateString(SourceName);
+  EpicsConnectionInfoBuilder MessageBuilder(builder);
+  if (!SourceName.empty()) {
+    MessageBuilder.add_source_name(Source);
+  }
+  MessageBuilder.add_timestamp(Timestamp);
+  MessageBuilder.add_type(Status);
+  builder.Finish(MessageBuilder.Finish(), "ep00");
+  DataSize = builder.GetSize();
+  auto RawBuffer = std::make_unique<std::int8_t[]>(DataSize);
+  std::memcpy(RawBuffer.get(), builder.GetBufferPointer(), DataSize);
+  return RawBuffer;
+}
+
+class Schema_ep00 : public ::testing::Test {
+public:
+  void SetUp() override {
+    try {
+      FileWriter::FlatbufferReaderRegistry::Registrar<
+          FileWriter::Schemas::ep00::FlatbufferReader>
+          RegisterIt("ep00");
+    } catch (...) {
+    }
+    File = hdf5::file::create(TestFileName, hdf5::file::AccessFlags::TRUNCATE);
+    RootGroup = File.root();
+    UsedGroup = RootGroup.create_group(NXLogGroup);
+  }
+  void TearDown() override { File.close(); };
+
+  std::string NXLogGroup{"SomeParentName"};
+  std::string TestFileName{"SomeTestFile.hdf5"};
+  hdf5::file::File File;
+  hdf5::node::Group RootGroup;
+  hdf5::node::Group UsedGroup;
+  SharedLogger Logger = getLogger();
+};
+
+// cppcheck-suppress syntaxError
+TEST_F(Schema_ep00, FileInitOk) {
+  ep00::HDFWriterModule Writer;
+  EXPECT_TRUE(Writer.init_hdf(UsedGroup, "{}") ==
+              HDFWriterModule_detail::InitResult::OK);
+  ASSERT_TRUE(RootGroup.has_group(NXLogGroup));
+  auto TestGroup = RootGroup.get_group(NXLogGroup);
+  EXPECT_TRUE(TestGroup.has_dataset(StatusName));
+  EXPECT_TRUE(TestGroup.has_dataset(TimestampName));
+}
+
+TEST_F(Schema_ep00, ReopenFile) {
+  ep00::HDFWriterModule Writer;
+  EXPECT_FALSE(Writer.reopen(UsedGroup) ==
+               HDFWriterModule_detail::InitResult::OK);
+}
+
+TEST_F(Schema_ep00, FileInitFail) {
+  ep00::HDFWriterModule Writer;
+  EXPECT_TRUE(Writer.init_hdf(UsedGroup, "{}") ==
+              HDFWriterModule_detail::InitResult::OK);
+  EXPECT_FALSE(Writer.init_hdf(UsedGroup, "{}") ==
+               HDFWriterModule_detail::InitResult::OK);
+}
+
+TEST_F(Schema_ep00, ReopenFileSuccess) {
+  ep00::HDFWriterModule Writer;
+  EXPECT_TRUE(Writer.init_hdf(UsedGroup, "{}") ==
+              HDFWriterModule_detail::InitResult::OK);
+  EXPECT_TRUE(Writer.reopen(UsedGroup) ==
+              HDFWriterModule_detail::InitResult::OK);
+}
+
+TEST_F(Schema_ep00, WriteDataSuccess) {
+
+  size_t BufferSize;
+  uint64_t Timestamp = 5555555;
+  std::string SourceName = "SIMPLE:DOUBLE";
+  auto Status = EventType::CONNECTED;
+  std::unique_ptr<std::int8_t[]> Buffer =
+      GenerateFlatbufferData(BufferSize, Timestamp, Status, SourceName);
+  ep00::HDFWriterModule Writer;
+  {
```

Comment:
This is repeating the expectations from FileInitOk and ReopenFileSuccess.
That is not usually a good thing.